src/root/WithdrawManager.ts

```diff
@@ -122,61 +140,45 @@ export default class WithdrawManager extends ContractsBase {
   }
 
-  async startExitForMintWithTokenURITokens(burnERC20TxHash, options?) {
-    const { payload, mint } = await this._startExitForMintWithTokenURITokens(
-      burnERC20TxHash,
-    )
-    return this.web3Client.send(
-      this.erc721Predicate.methods.startExitForMintWithTokenURITokens(
-        payload,
-        mint,
-      ),
-      options,
-    )
+  async startExitWithBurntERC721Tokens(burnTxHash, options?) {
+    // @todo
   }
 
-  async startBulkExitForMintWithTokenURITokens(burnTxs, options?) {
-    const payloads = []
-    const mints = []
-    // note that these calls will be simultaneous
-    await bluebird.map(burnTxs, async tx => {
-      const { payload, mint } = await this._startExitForMintWithTokenURITokens(
-        tx,
-      )
-      payloads.push(payload)
-      mints.push(mint)
-    })
-    return this.web3Client.send(
-      this.erc721Predicate.methods.startBulkExitForMintWithTokenURITokens(
-        ethUtils.rlp.encode(payloads),
-        ethUtils.rlp.encode(mints),
-      ),
-      options,
-    )
+  /**
+   * Start an exit for a token with metadata (token uri) that was minted and burnt on the side chain
+   * Wrapper over contract call: [MintableERC721Predicate.startExitForMetadataMintableBurntToken](https://github.com/maticnetwork/contracts/blob/e2cb462b8487921463090b0bdcdd7433db14252b/contracts/root/predicates/MintableERC721Predicate.sol#L66)
+   * @param burnTxHash Hash of the burn transaction on Matic
+   * @param predicate address of MintableERC721Predicate
+   */
+  async startExitForMetadataMintableBurntToken(burnTxHash, predicate: address, options?) {
+    const { payload, mint } = await this._buildPayloadAndFindMintTransaction(burnTxHash)
+    const _predicate = new this.web3Client.parentWeb3.eth.Contract(MintableERC721PredicateArtifact.abi, predicate)
+    const txObject = _predicate.methods.startExitForMetadataMintableBurntToken(payload, mint)
+    const _options = await this._fillOptions(options, txObject, this.web3Client.parentWeb3)
+    return this.web3Client.send(txObject, _options)
   }
 
-  private async _startExitForMintWithTokenURITokens(burnTxHash, options?) {
+  private async _buildPayloadAndFindMintTransaction(burnTxHash) {
     const payload = await this._buildPayloadForExit(burnTxHash)
-    const burnReceipt = await this.web3Client
-      .getMaticWeb3()
-      .eth.getTransactionReceipt(burnTxHash)
+    const burnReceipt = await this.web3Client.web3.eth.getTransactionReceipt(burnTxHash)
     const withdrawEvent = burnReceipt.logs.find(
       l => l.topics[0].toLowerCase() === WithdrawManager.WITHDRAW_EVENT_SIG,
     )
     const tokenId = withdrawEvent.data
-    const mintEvents = await this.getERC721TokenContract(
-      burnReceipt.to,
-    ).getPastEvents('Transfer', {
+    logger.debug({ burnTxHash, burnReceipt, withdrawEvent, tokenId })
+    const contract = new this.web3Client.web3.eth.Contract(ChildERC721MintableArtifact.abi, burnReceipt.to)
```

Comment:
I think here `web3` object should be `parentWeb3`
It is looking for events on `childChain`, that's why always returning empty result for `getPastEvents`
@atvanguard